triggers.md

```diff
@@ -1,3 +1,133 @@
+```
+
+The rules are divided into sections by the action, such as "dns-prefetch", "preconnect", "prefetch" or "prerender", that they authorize. A user agent may always choose to stop early, for instance by prefetching where prerendering was permitted.
+
+Since these are rules for optional behavior (the UA is always free not to speculate), parsing of these can be somewhat conservative. If a UA sees a rule it does not understand, that rule is discarded. Because a rule always authorizes additional speculation (but does not confine speculation allowed by other rules), discarding rules never authorizes more speculation than the author intended, only less.
+
+The rules are divided into a number of simple pieces that build on one another, and could be shipped largely independently, due to the conservative parsing strategy.
+
+### Rules
+
+A rule has a _source_ which identifies the URLs to which the rule applies.
+
+A rule may include a _score_ between 0.0 and 1.0 (inclusive), defaulting to 0.5, which is a hint about how likely the user is to navigate to the URL. It is expected that UAs will treat this monotonically (i.e., all else equal, increasing the score associated with a rule will make the UA speculate no less than before for that URL, and decreasing the score will not make the UA speculate where it previously did not). However, the user agent may select a link with a lower author-assigned score than another if its heuristics suggest it is a better choice.
+
+#### List rules
+
+A list rule must have an express list of the _URLs_ to which the rule applies, even if no link or other reference to the URL exists in the document. This is especially useful in cases where the expected navigation corresponds to a link that will be added dynamically, or to an anticipated script-initiated navigation.
+
+These URLs will be parsed relative to the document base URL (if inline in a document) or relative to the external resource URL (if externally fetched).
+
+#### Document rules
+
+A document rule allows the user agent to find URLs for speculation from link elements in the page. It may include criteria which restrict which of these links can be used.
+
+The URL can be compared against [URL patterns][urlpattern] (parsed relative to the same base URL as URLs in list rules).
+
+<dl>
+<dt><code>"if_href_matches": [...]</code></dt>
+<dd>requires that the link URL match at least one pattern from the list</dd>
+<dt><code>"if_not_href_matches": [...]</code></dt>
+<dd>requires that the link URL not match any pattern from the list</dd>
+</dl>
+
+The link element itself can also be [matched][selector-match] using [CSS selectors][selectors].
+
+<dl>
+<dt><code>"if_selector_matches": [...]</code></dt>
+<dd>requires that the link element match at least one selector from the list</dd>
+<dt><code>"if_not_selector_matches": [...]</code></dt>
+<dd>requires that the link element not match any selector from the list</dd>
+</dl>
+
+#### Possible future extension: Requirements
```

Comment:
I feel like this and other sections might not be nested under "Rules"? Certainly starting at "Proposed processing model".

Reply:
Lifted proposed processing model & dev tools